### ISSUE TITLE
Tell Bio-Formats to just open one file.

### ISSUE THREAD

**Issue by tischi** [BIOP/bigdataviewer-image-loaders#12]
@NicoKiaru @joshmoore @dgault

Could we add an option to `OpenerSettings` to tell `Bio-Formats` to just open the one file that one gives the path to?

There is this issue that when one tries to open a TIFF file in a folder, then Bio-Formats tries to figure out whether this is part of some sequence and then always opens all the TIFF files that belong to that sequence. Sometimes this can be nice, but sometimes not.
It would really be great (for me essential) to control this.

**Comment by joshmoore** [comment]
@dgault can correct me, but from the Bio-Formats side, I don't think there's a general way to say, "only open this one file" unless you can pre-detect the file format. If you know for example that you want `TIFFReader` rather than `OMETiffReader` to get the "just open this file" behavior, then you can instantiate that rather than `ImageReader` which tries all possible readers (and/or use a readers.txt which does the same). I'm a `OpenerSettings` n00b but the only other option I can think of is to say, "don't use Bio-Formats".

**Comment by NicoKiaru** [comment]
@tischi In opener settings, everything is done through bio-formats API (you can only specify a series). If there's a bio-formats way to do what you want, then we can try to implement it here. There's still an 'opt' field in opener-settings which can be potentially used for that. But we need a functioning code in bio-formats ( independently of bdv-image-loaders)

**Comment by tischi** [comment]
@NicoKiaru could we add an option `useTIFFReader( boolean useTIFFReader )`?

**Comment by NicoKiaru** [comment]
Maybe, but it's going to be a pain.

It involves changing:

https://github.com/BIOP/bigdataviewer-image-loaders/blob/93c22096aa267828c7a314ab430a0eb5a2da5800/src/main/java/ch/epfl/biop/bdv/img/bioformats/BioFormatsOpener.java#L336

and

https://github.com/BIOP/bigdataviewer-image-loaders/blob/93c22096aa267828c7a314ab430a0eb5a2da5800/src/main/java/ch/epfl/biop/bdv/img/bioformats/BioFormatsHelper.java#L57

And making sure the right parameter is passed to the right argument and maintaining compatibility with already serialized datasets.

Can you just check whether it gives the right behaviour on one of your file ?

**Comment by tischi** [comment]
> Can you just check whether it gives the right behaviour on one of your file ?

Good idea. I will do this and then get back to you.

**Comment by NicoKiaru** [comment]
@tischi this issue is stalled... I guess you found a way around it. I'm closing it. Please re-open if needed

**Comment by tischi** [comment]
Yeah, I think if there's TIFF files, by default I am now using something else than Bio-Formats to open it.
